src/OpenSpaceToolkit/Astrodynamics/Dynamics/Thruster.cpp

```diff
@@ -83,6 +105,11 @@ VectorXd Thruster::computeContribution(
     const Instant& anInstant, const VectorXd& x, const Shared<const Frame>& aFrameSPtr
 ) const
 {
+    if (applicabilityInterval_.isDefined() && !applicabilityInterval_.contains(anInstant))
```

Comment:
I think this should be done at a guidance law level rather than the thruster level, thrusters should stay agnostic to any interval.